edge-modules/api-proxy-module/src/monitors/config_monitor.rs

```diff
@@ -96,8 +105,8 @@ pub fn start(
     Ok((monitor_loop, shutdown_handle))
 }
 
-fn save_raw_config(twin: &TwinProperties) -> Result<()> {
-    let json = twin.properties.get_key_value(PROXY_CONFIG_TAG);
+fn save_raw_config(twin: &std::collections::HashMap<String, serde_json::Value>) -> Result<()> {
+    let json = twin.get_key_value(PROXY_CONFIG_TAG);
 
     //Get value associated with the key and extract is as a string.
     let str = (*(json
```

Comment:
this code looks scary :)
```rust
fn save_raw_config(twin: &std::collections::HashMap<String, serde_json::Value>) -> Result<()> {
    let config = twin
        .get(PROXY_CONFIG_TAG)
        .ok_or_else(|| anyhow!("Key {} not found in twin", PROXY_CONFIG_TAG))?;

    let config = config
        .as_str()
        .ok_or_else(|| anyhow!("Cannot extract json as base64 string"))?;

    let bytes =
        base64::decode(config).map_err(|err| anyhow!("Cannot decode base64. Caused by {}", err))?;

    file::write_binary_to_file(&bytes, PROXY_CONFIG_PATH_RAW).map_err(|err| {
        anyhow!(
            "Cannot write config file to path: {}. Caused by {}",
            PROXY_CONFIG_PATH_RAW,
            err
        )
    })?;

    Ok(())
}

```

Reply:
I am not still learning canonical rust, is there a proper way of doing this?